ui/lib/core/addon/utils/client-count-utils.js

```diff
@@ -87,3 +71,89 @@ export const homogenizeClientNaming = (object) => {
   }
   return object;
 };
+
+const flattenDataset = (object) => {
+  if (Object.keys(object).includes('counts') && object.counts) {
+    let flattenedObject = {};
+    Object.keys(object['counts']).forEach((key) => (flattenedObject[key] = object['counts'][key]));
+    return homogenizeClientNaming(flattenedObject);
+  }
+  return object;
+};
+
+const sortMonthsByTimestamp = (monthsArray) => {
+  // backend is working on a fix to sort months by date
+  // right now months are ordered in descending client count number
+  const sortedPayload = [...monthsArray];
+  return sortedPayload.sort((a, b) =>
+    compareAsc(parseAPITimestamp(a.timestamp), parseAPITimestamp(b.timestamp))
+  );
+};
+
+const namespaceArrayToObject = (totalClientsByNamespace, newClientsByNamespace, month) => {
+  // first nest 'new_clients' within each namespace and mount object respectively
+  totalClientsByNamespace.forEach((ns) => {
```

Comment:
If we're mutating the data, it would be better to return a map of the `totalClientsByNamespace` instead. So eg 
```suggestion
  return totalClientsByNamespace.map((ns) => {
```
and then rather than deleting keys you can only include the items you want.

Reply:
I cleaned up the comment a bit. Do you think it still makes sense to map, as I'm only manipulating the data to construct and then return the new object `namespaces_by_key` below. Happy to refactor accordingly! Just want to make sure we're on the same page.